packages/devtools_app/lib/src/screens/inspector/inspector_controller.dart

```diff
@@ -194,46 +191,46 @@ class InspectorController extends DisposableController
   final bool isSummaryTree;
 
   /// Parent InspectorController if this is a details subtree.
-  InspectorController parent;
+  InspectorController? parent;
 
-  InspectorController details;
+  InspectorController? details;
 
   InspectorTreeController inspectorTree;
   final FlutterTreeType treeType;
 
   bool _disposed = false;
 
-  RateLimiter _refreshRateLimiter;
+  late RateLimiter _refreshRateLimiter;
 
   /// Groups used to manage and cancel requests to load data to display directly
   /// in the tree.
-  InspectorObjectGroupManager _treeGroups;
+  InspectorObjectGroupManager? _treeGroups;
 
   /// Groups used to manage and cancel requests to determine what the current
   /// selection is.
   ///
   /// This group needs to be kept separate from treeGroups as the selection is
   /// shared more with the details subtree.
   /// TODO(jacobr): is there a way we can unify the selection and tree groups?
-  InspectorObjectGroupManager _selectionGroups;
+  InspectorObjectGroupManager? _selectionGroups;
```

Comment:
assigned to null in dispose

Reply:
add  non-nullable getter. otherwise we are polluting the code with null checks that are only needed if there is use after dispose.

Reply:
see https://github.com/flutter/devtools/pull/3928#discussion_r838787666